src/ucp/amo/amo.inl

```diff
@@ -59,22 +60,23 @@ UCP_PROGRESS_AMO_DECL(uint64_t, uct_ep_atomic_fadd64, UCP_AMO_ONE_PARAM)
 UCP_PROGRESS_AMO_DECL(uint64_t, uct_ep_atomic_cswap64, UCP_AMO_TWO_PARAM)
 
 #define UCP_POST_AMO_DECL(_type, _function) \
-static ucs_status_t _UCP_PROGRESS_AMO_NAME(_function)(uct_pending_req_t *_self) \
+    static ucs_status_t _UCP_PROGRESS_AMO_NAME(_function)(uct_pending_req_t *_self) \
     { \
         ucp_request_t *req = ucs_container_of(_self, ucp_request_t, send.uct); \
         ucp_rkey_h rkey    = req->send.amo.rkey; \
         ucp_ep_t *ep       = req->send.ep; \
         _type value = (_type)req->send.amo.value;         \
         uint64_t remote_addr = req->send.amo.remote_addr; \
         ucs_status_t status; \
-        uct_rkey_t uct_rkey; \
-        uint32_t lane; \
         \
-        UCP_EP_RESOLVE_RKEY_AMO(ep, rkey, req->send.lane, uct_rkey, \
-                                return UCS_ERR_UNREACHABLE); \
-        lane = req->send.lane; \
-        status = UCS_PROFILE_CALL(_function, ep->uct_eps[lane], value,\
-                                  remote_addr, uct_rkey); \
+        status = UCP_RKEY_RESOLVE(rkey, ep, amo); \
+        if (status != UCS_OK) { \
```

Comment:
not needed here because UCP_RKEY_RESOLVE returns a value which is known in compile-time, and it already uses ucs_unlikely()